src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -68,10 +68,13 @@ public DocumentExecuter(IDocumentBuilder documentBuilder, IDocumentValidator doc
         private ExecutionDelegate BuildExecutionDelegate(IEnumerable<IConfigureExecution> configurations)
         {
             ExecutionDelegate execution = CoreExecuteAsync;
-            var configurationArray = configurations.ToArray();
-            for (var i = configurationArray.Length - 1; i >= 0; i--)
+
+            // OrderBy here performs a stable sort; that is, if the sort order of two elements are equal,
+            // the order of the elements are preserved. The order is reversed since each execution wraps
+            // the prior configured executions. OrderByDescending is not used because that would result
```

Comment:
Do you mean `OrderByDescending ` is not stable?

Reply:
No, it is stable, which produces different results compared to ascending reversed when it encounters entries with the same sort value.

Assuming B1 == B2:

| Original | Ascending | Ascending Reversed | Descending |
|-|-|-|-|
| A, D, B1, B2 | A, B1, B2, D | D, B2, B1, A | D, B1, B2, A |
| A, D, B2, B1 | A, B2, B1, D | D, B1, B2, A | D, B2, B1, A |

In this case we require a stable ascending sort, reversed.  Descending will not do.  Note that MS documented the fact that OrderBy performs a stable sort, so we can rely on this.

If you have a better way of explaining this, please suggest a revision to the comment - thanks!

Reply:
understood